File: namsor/java/src/main/java/oseg/namsor_examples/GenderJapaneseNamePinyinBatchExample.java
package oseg.namsor_examples;

import app.namsor.client.ApiException;
import app.namsor.client.Configuration;
import app.namsor.client.api.*;
import app.namsor.client.auth.*;
import app.namsor.client.JSON;
import app.namsor.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class GenderJapaneseNamePinyinBatchExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        config.setApiKey("YOUR_API_KEY");

        var personalNames1 = new FirstLastNameIn();
        personalNames1.id("id");
        personalNames1.firstName("firstName");
        personalNames1.lastName("lastName");

        var personalNames2 = new FirstLastNameIn();
        personalNames2.id("id");
        personalNames2.firstName("firstName");
        personalNames2.lastName("lastName");

        var personalNames = new ArrayList<FirstLastNameIn>(List.of (
            personalNames1,
            personalNames2
        ));

        var batchFirstLastNameIn = new BatchFirstLastNameIn();
        batchFirstLastNameIn.personalNames(personalNames);

        try
        {
            var response = new JapaneseApi(config).genderJapaneseNamePinyinBatch(
                batchFirstLastNameIn
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling JapaneseApi#genderJapaneseNamePinyinBatch");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: namsor/kotlin/src/main/kotlin/oseg/namsor_examples/GenderJapaneseNamePinyinBatchExample.kt
package oseg.namsor_examples

import app.namsor.client.infrastructure.*
import app.namsor.client.apis.*
import app.namsor.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class GenderJapaneseNamePinyinBatchExample
{
    fun genderJapaneseNamePinyinBatch()
    {
        ApiClient.apiKey["X-API-KEY"] = "YOUR_API_KEY"

        val personalNames1 = FirstLastNameIn(
            id = "id",
            firstName = "firstName",
            lastName = "lastName",
        )

        val personalNames2 = FirstLastNameIn(
            id = "id",
            firstName = "firstName",
            lastName = "lastName",
        )

        val personalNames = arrayListOf<FirstLastNameIn>(
            personalNames1,
            personalNames2,
        )

        val batchFirstLastNameIn = BatchFirstLastNameIn(
            personalNames = personalNames,
        )

        try
        {
            val response = JapaneseApi().genderJapaneseNamePinyinBatch(
                batchFirstLastNameIn = batchFirstLastNameIn,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling JapaneseApi#genderJapaneseNamePinyinBatch")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling JapaneseApi#genderJapaneseNamePinyinBatch")
            e.printStackTrace()
        }
    }
}
